html/modules/custom/reliefweb_import/src/Command/ReliefwebImportCommand.php

```diff
@@ -0,0 +1,680 @@
+  }
+
+  /**
+   * Sanitize text, converting it to markdown.
+   *
+   * @param string $field
+   *   Field name.
+   * @param string $text
+   *   Field text content.
+   * @param string $format
+   *   Format to which conver the text if not already HTML.
+   *
+   * @return string
+   *   Sanitized content.
+   */
+  protected function sanitizeText($field, $text, $format = 'plain_text') {
+    if (!is_string($text)) {
+      return '';
+    }
+
+    // Trim the input text.
+    $text = trim($text);
+    if (empty($text)) {
+      return '';
+    }
+
+    // Sometimes the parser bugs and leaves the closing field tag at the end
+    // of the text. We remove it before proceeding.
+    $position = mb_strpos($text, '</' . $field . '>');
+    if ($position !== FALSE) {
+      $text = mb_substr($text, 0, $position);
+    }
+
+    // Clean the text, removing notably control characters.
+    $text = $this->cleanText($text);
+
+    // Check if the text is wrapped in a CDATA.
+    if (mb_stripos($text, '<![CDATA[') === 0) {
+      $end = mb_strpos($text, ']]>');
+      $text = mb_substr($text, 9, $end !== FALSE ? $end - 9 : NULL);
+    }
+
+    // Check if the content contains some non encoded html tags, in which case
+    // we will assume that the text is non encoded html/markdown. For that we
+    // simply search for a closing tag '</...>'. Otherwise we decode the text.
+    if (preg_match('#</[^>]+>#', $text) !== 1) {
+      $text = html_entity_decode($text, ENT_QUOTES, 'UTF-8');
+    }
+
+    // Convert the text to HTML.
+    /*
+    if (function_exists('check_markup')) {
+    $text = check_markup($text, $format)->__toString();
+    }
+     */
+
+    // We then sanitize the HTML string.
+    $sanitizer = new HtmlSanitizer();
+    $text = $sanitizer->sanitizeHtml($text);
+
+    // Remove embedded content.
+    $text = $this->stripEmbeddedContent($text);
+
+    // Finally we convert it to markdown.
+    $converter = new HtmlConverter();
+    $converter->getConfig()->setOption('strip_tags', TRUE);
+    $converter->getConfig()->setOption('use_autolinks', FALSE);
+    $converter->getConfig()->setOption('header_style', 'atx');
+    $converter->getConfig()->setOption('strip_placeholder_links', TRUE);
+
+    $text = trim($converter->convert($text));
+
+    return $text;
+  }
+
+  /**
+   * Clean a text.
+   *
+   * 1. Replace tabulations with double spaces.
+   * 2. Replace non breaking spaces with normal spaces.
+   * 3. Remove control characters (except line feed).
+   * 4. Optionally, replace line breaks and consecutive whitespaces.
+   * 4. Trim the text.
+   *
+   * @param string $text
+   *   Text to clean.
+   * @param array $options
+   *   Associative array with the following replacement options:
+   *   - line_breaks (boolean): replace line breaks with spaces.
+   *   - consecutive (boolean): replace consecutive whitespaces.
+   *
+   * @return string
+   *   Cleaned text.
+   */
+  protected function cleanText($text, array $options = []) {
```

Comment:
For info, there is new utility helper with the same code: https://github.com/UN-OCHA/rwint9-site/blob/dea62fca917f7e9c1e16d472e4be4ee51d1b7a6c/html/modules/custom/reliefweb_utility/src/Helpers/TextHelper.php#L30